crates/client/src/api.rs

```diff
@@ -60,41 +123,22 @@ impl Client {
     }
 
     /// Gets the latest checkpoint from the registry.
-    pub async fn latest_checkpoint(&self) -> Result<SerdeEnvelope<MapCheckpoint>, ClientError> {
+    pub async fn latest_checkpoint(&self) -> ApiResult<CheckpointResponse> {
         let url = self.0.join("fetch/checkpoint").unwrap();
         tracing::debug!("getting latest checkpoint at `{url}`");
-
-        let response = reqwest::get(url).await?;
-        if !response.status().is_success() {
-            return Err(ClientError::ApiError {
-                registry: self.0.host_str().unwrap_or("").to_string(),
-                status: response.status().as_u16(),
-                body: response.text().await?,
-            });
-        }
-
-        let response = response.json::<CheckpointResponse>().await?;
-        Ok(response.checkpoint)
+        into_result::<_, FetchError>(reqwest::get(url).await?).await
     }
 
     /// Fetches package log entries from the registry.
-    pub async fn fetch_logs(&self, request: FetchRequest) -> Result<FetchResponse, ClientError> {
+    pub async fn fetch_logs(&self, request: FetchRequest) -> ApiResult<FetchResponse> {
```

Comment:
Shouldn't this only ever return `FetchError`s as opposed to any `ApiError`? and similarly for the other api methods. If you decide you want to handle different fetch errors differently you have to match against the fetch variant which feels strange.